src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -126,10 +129,17 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
         this.etLogEventType = etLogEventType;
         this.eventTypeOptionsValidator = eventTypeOptionsValidator;
         this.adminService = adminService;
+        this.authorizationService = authorizationService;
     }
 
     public List<EventType> list() {
-        return eventTypeRepository.list();
+        return authorizationService.filter(eventTypeRepository.list()
+                .stream()
+                .map(EventType::asResource)
+                .collect(Collectors.toList()))
+                .stream()
+                .map(Resource<EventType>::get)
```

Comment:
I don't get this part. Why is it necessary to "unwrapp" the resource? I thought the API expected a `List<Resource<T>>`, no?